tests/test_h1_smart_device.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+from _testutils import MockHidapiDevice, Report
+
+from liquidctl.driver.hwmon import HwmonDevice
+from liquidctl.driver.smart_device import H1SmartDevice
+
+SAMPLE_STATUS = bytes.fromhex(
+    "75021320020d85bcabab94188f5f010000a00f0032020284021e1e02f9066464"
+    "0000000000000000000000000000000000000000000000000000000000000005"
+)
+
+
+class MockH1Smart(MockHidapiDevice):
+    def __init__(self, raw_speed_channels, raw_led_channels):
+        super().__init__()
+        self.raw_speed_channels = raw_speed_channels
+        self.raw_led_channels = raw_led_channels
+
+    def write(self, data):
+        reply = bytearray(64)
+        if data[0:2] == [0x10, 0x01]:
+            reply[0:2] = [0x11, 0x01]
+        elif data[0:2] == [0x20, 0x03]:
+            reply[0:2] = [0x21, 0x03]
+            reply[14] = self.raw_led_channels
+            if self.raw_led_channels > 1:
+                reply[15 + 1 * 6] = 0x10
+                reply[15 + 2 * 6] = 0x11
+        self.preload_read(Report(reply[0], reply[1:]))
+        return super().write(data)
+
+
+@pytest.fixture
+def mock_smart2():
+    raw = MockH1Smart(raw_speed_channels=2, raw_led_channels=0)
+    dev = H1SmartDevice(raw, "Mock H1 V2", speed_channel_count=2, color_channel_count=0)
+    dev.connect()
+    return dev
+
+
+@pytest.mark.parametrize("has_hwmon,direct_access", [(False, True)])
+def test_initializes(mock_smart2, has_hwmon, direct_access, tmp_path):
+    if has_hwmon:
+        mock_smart2._hwmon = HwmonDevice("mock_module", tmp_path)
+
+    _ = mock_smart2.initialize(direct_access=direct_access)
+
+    writes = len(mock_smart2.device.sent)
+    if not has_hwmon or direct_access:
+        assert writes == 4
+    else:
+        assert writes == 2
+
+
+@pytest.mark.parametrize("has_hwmon,direct_access", [(False, True)])
```

Comment:
Since we're going to override `get_status()` directly (see previous comment), I think you can remove this parametrization (please rename the function accordingly).